tests/testthat/out/error-index-type.txt

```diff
@@ -0,0 +1,24 @@
+> vec_as_index(mtcars, 10L)
+Error: Must subset with positions or names.
+x `i` has the wrong type `data.frame<
```

Comment:
It occurs to me that you only really need to show the details of the data frame if the expected type is another data frame. i.e. we only need to conditionally expand the details of the type.

```
vec_assert(data.frame(), integer())
#> Expected <integer> got <data.frame>

vec_assert(data.frame(x = integer()), data.frame(y = integer())
#> Expected <data.frame<x: integer>> got <data.frame<y: integer>>
```

We don't need to do anything about that now, but it's worth considering that how we display the actual type might depend on the expected type.

Reply:
Great idea, I have opened an issue: https://github.com/r-lib/vctrs/issues/609